src/main.rs

```diff
@@ -113,6 +113,20 @@ fn main() {
         _ => atty::is(Stream::Stdout),
     };
 
+    let path_separator: Option<u8> = match matches.value_of("path-separator") {
+        Some(sep_str) => {
+            let sep = sep_str.as_bytes();
+            if sep.len() != 1 {
+                print_error_and_exit!(
+                    "'{}' is not a valid path separator. See 'fd --help'.",
+                    sep_str
+                );
+            }
+            Some(sep[0])
+        }
+        None => None,
```

Comment:
Actually, I think that case can not happen (since you require `--path-separator` to have a value). If this is the case, I think you can simply unwrap the Option:
```
matches.value_of("path-separator").except("path-separator")
```

Reply:
I confirm that not passing the argument triggers the None case.